On Solaris pyconfig.h is delivered in the normal python header directory.
As this is Solaris-specific, it is not suitable for upstream.

--- Python-3.4.0/Lib/distutils/sysconfig.py.~1~	2014-03-16 19:31:29.000000000 -0700
+++ Python-3.4.0/Lib/distutils/sysconfig.py	2014-03-17 13:50:16.261084406 -0700
@@ -242,7 +242,7 @@
         else:
             inc_dir = _sys_home or project_base
     else:
-        inc_dir = get_python_inc(plat_specific=1)
+        inc_dir = get_python_inc(plat_specific=0)
     if get_python_version() < '2.2':
         config_h = 'config.h'
     else:
